zingolib/src/wallet/keys/unified.rs

```diff
@@ -352,6 +354,25 @@ impl WalletCapability {
         Ok(Self::new_from_seed(config, &bip39_seed, position))
     }
 
+    /// Creates a new `WalletCapability` from a unified spending key.
+    pub fn new_from_usk(usk: &[u8]) -> Result<Self, String> {
+        // Decode unified spending key
+        let usk = UnifiedSpendingKey::from_bytes(Era::Orchard, usk)
+            .map_err(|_| "Error decoding unified spending key.")?;
+
+        // Workaround https://github.com/zcash/librustzcash/issues/929 by serializing and deserializing the transparent key.
```

Comment:
I did a bit of digging, and it turns out the `TransparentPrivKey` type being used here is actually a zingolib type, albeit a redefinition/copy-paste of the `hdwallet` type with a bunch of additional methods....I think it would make more sense to use the `hdwallet` type (either sacrifice a little bit of method call syntax and define local helpers instead of being method calls, or just define a local trait with the additional functionality in order to keep the method syntax) to avoid the surprising amount of duplicated code I just discovered, but that's not a blocking concern and definitely beyond the scope of this PR.

Reply:
Where's the `hdwallet` type defined?   Why do we have to copy paste?

Reply:
IIRC it was because the struct we might have reused carried private fields, and exposed them only in a limited way.
But as I think @AloeareV was alluding to, I may have even been looking at a similarly named, but incompatible type anyway, so maybe this was all necessary. I'd have to go back and check.